server/server_test.go

```diff
@@ -237,10 +239,13 @@ func (cli *testServerClient) runTestPrepareResultFieldType(t *testing.T) {
 	var param int64 = 83
 	cli.runTests(t, nil, func(dbt *testkit.DBTestKit) {
 		stmt, err := dbt.GetDB().Prepare(`SELECT ?`)
+		defer func() {
+			err = stmt.Close()
+			require.NoError(t, err)
+		}()
```

Comment:
I believe you should put this at L243, the original position. `stmt` may be `nil` if `err` is not nil. The current logic works because the program will fatal if `err` is not nill. If it is not fatal, the deferred function will panic.

Reply:
Fixed